pkg/operator/operator_test.go

```diff
@@ -2,67 +2,124 @@ package operator
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/informers"
-	"k8s.io/client-go/kubernetes"
+	fakekube "k8s.io/client-go/kubernetes/fake"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/client-go/util/workqueue"
 
 	v1 "github.com/openshift/api/config/v1"
 	fakeos "github.com/openshift/client-go/config/clientset/versioned/fake"
 	configinformersv1 "github.com/openshift/client-go/config/informers/externalversions"
-	appsv1 "k8s.io/api/apps/v1"
-	fakekube "k8s.io/client-go/kubernetes/fake"
+	"github.com/stretchr/testify/assert"
 )
 
+const (
+	deploymentName   = "machine-api-controllers"
+	targetNamespace  = "test-namespace"
+	hcControllerName = "machine-healthcheck-controller"
+)
+
+func newFeatureGate(featureSet v1.FeatureSet) *v1.FeatureGate {
+	return &v1.FeatureGate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: MachineAPIFeatureGateName,
+		},
+		Spec: v1.FeatureGateSpec{
+			FeatureSet: featureSet,
+		},
+	}
+}
+
+func newOperatorConfig() *OperatorConfig {
+	return &OperatorConfig{
+		targetNamespace,
+		Controllers{
+			"docker.io/openshift/origin-aws-machine-controllers:v4.0.0",
+			"docker.io/openshift/origin-machine-api-operator:v4.0.0",
+			"docker.io/openshift/origin-machine-api-operator:v4.0.0",
+		},
+	}
+}
+
+func newFakeOperator(kubeObjects []runtime.Object, osObjects []runtime.Object, stopCh <-chan struct{}) *Operator {
+	kubeClient := fakekube.NewSimpleClientset(kubeObjects...)
+	osClient := fakeos.NewSimpleClientset(osObjects...)
+	kubeNamespacedSharedInformer := informers.NewSharedInformerFactoryWithOptions(kubeClient, 2*time.Minute, informers.WithNamespace(targetNamespace))
+	configSharedInformer := configinformersv1.NewSharedInformerFactoryWithOptions(osClient, 2*time.Minute)
+	featureGateInformer := configSharedInformer.Config().V1().FeatureGates()
+	deployInformer := kubeNamespacedSharedInformer.Apps().V1().Deployments()
+
+	optr := &Operator{
+		kubeClient:             kubeClient,
+		osClient:               osClient,
+		featureGateLister:      featureGateInformer.Lister(),
+		deployLister:           deployInformer.Lister(),
+		imagesFile:             "fixtures/images.json",
+		namespace:              targetNamespace,
+		eventRecorder:          record.NewFakeRecorder(50),
+		queue:                  workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "machineapioperator"),
+		deployListerSynced:     deployInformer.Informer().HasSynced,
+		featureGateCacheSynced: featureGateInformer.Informer().HasSynced,
+	}
+
+	configSharedInformer.Start(stopCh)
+	kubeNamespacedSharedInformer.Start(stopCh)
+
+	optr.syncHandler = optr.sync
+	deployInformer.Informer().AddEventHandler(optr.eventHandler())
+	featureGateInformer.Informer().AddEventHandler(optr.eventHandler())
+
+	return optr
+}
+
 // TestOperatorSync_NoOp tests syncing to ensure that the mao reports available
 // for platforms that are no-ops.
 func TestOperatorSync_NoOp(t *testing.T) {
 	cases := []struct {
-		platform     configv1.PlatformType
```

Comment:
for future PRs of this size it be nice if we stick to the necessary changes so avoid unnecessary changes for the sake of easier review and smoother workflow

Reply:
it necessary change, because this unittest worked before for cases when the deployment manifests does not exist, but not we do not depend on the deployment manifest, so I needed to change the test

Reply:
sorry I meant the import names, e.g `configv1 <->  v1`